Content.Server/Ninja/Systems/NinjaSystem.cs

```diff
@@ -0,0 +1,678 @@
+using Content.Server.Mind.Components;
+using Content.Server.Objectives;
+using Content.Server.Popups;
+using Content.Server.Power.Components;
+using Content.Server.PowerCell;
+using Content.Server.Traitor;
+using Content.Server.Warps;
+using Content.Shared.Actions;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Alert;
+using Content.Shared.Damage.Components;
+using Content.Shared.Database;
+using Content.Shared.Doors.Components;
+using Content.Shared.Emag.Systems;
+using Content.Shared.Examine;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Ninja.Systems;
+using Content.Shared.Popups;
+using Content.Shared.PowerCell.Components;
+using Content.Shared.Research.Components;
+using Content.Shared.Roles;
+using Content.Shared.Rounding;
+using Content.Shared.Stealth.Components;
+using Content.Shared.Tag;
+using Content.Shared.Weapons.Melee.Events;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace Content.Server.Ninja.Systems;
+
+public sealed partial class NinjaSystem : SharedNinjaSystem
+{
+    [Dependency] private readonly ActionsSystem _actions = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly IChatManager _chatMan = default!;
+    [Dependency] private readonly DoAfterSystem _doafter = default!;
+    [Dependency] private readonly ElectrocutionSystem _electrocution = default!;
+    [Dependency] private readonly EmagSystem _emag = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly SharedHandsSystem _hands = default!;
+    [Dependency] private readonly SharedSubdermalImplantSystem _implants = default!;
+    [Dependency] private readonly PopupSystem _popups = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly PowerCellSystem _powerCell = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly TagSystem _tags = default!;
+    [Dependency] private readonly TraitorRuleSystem _traitorRule = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    private readonly HashSet<SpaceNinjaComponent> _activeNinja = new();
```

Comment:
Please don't store data on systems as it doesn't handle serialization or replays properly.

Reply:
i was copying door system i think, how would i do updating ninja components without it

Reply:
oh right EntityQuery duh

also gonna make issue to update doors since thats smelly